src/utp_wrapper.rs

```diff
@@ -34,11 +38,16 @@ impl UtpWrapper {
                     socket.set_read_timeout(Some(CHECK_FOR_NEW_WRITES_INTERVAL_MS));
                     'outer: loop {
                         let mut buf = [0; BUFFER_SIZE];
-                        match socket.recv_from(&mut buf) {
+                        match socket.recv_from(&mut buf[..]) {
                             Ok((0, _src)) => break,
                             Ok((amt, _src)) => {
                                 let buf = &buf[..amt];
-                                let _ = input_tx.send(Vec::from(buf));
+                                match input_tx.send(Vec::from(buf)) {
```

Comment:
That's a bit silly. I like to see exactly what data is getting thrown away. `if let` obscures that.